### PR TITLE
Feat/window state/custom state filename

### PR DESCRIPTION
**Pull request by thewh1teagle**
Depends on https://github.com/tauri-apps/plugins-workspace/pull/1078
Add option to use custom filename for window state file with the method `with_filename`

### CODE REVIEW

**Comment by amrbashir** [comment]
Can you elaborate why you need this? Also see prior discussion here https://github.com/tauri-apps/plugins-workspace/issues/516

**Comment by ImageDeeply** [comment]
Hello. I work with @thewh1teagle 

I don't mind having a hidden file with pre-defined name for binary data, but I think a json file should be visible and end with '.json'. If everyone agrees, then that suffices. But I assume that some people want to preserve the leading '.' and/or not append '.json' (and/or have other reasons for a custom filename). Hence this PR.

More generally, while I appreciate the benefits of 'opinionated' tools, I'm one of those app developers who want lots of choices so we can make our own tradeoffs.

Either way, thanks for considering.

File: plugins/window-state/src/lib.rs [hunk 1]
```diff
@@ -24,12 +24,15 @@ use std::{
...
+    sync::{Arc,Mutex,OnceLock},
 };
 
 mod cmd;
 
-pub const STATE_FILENAME: &str = ".window-state";
+pub const DEFAULT_STATE_FILENAME: &str = ".window-state";
+static STATE_FILENAME: OnceLock<String> = OnceLock::new();
+
+static USE_JSON: OnceLock<bool> = OnceLock::new();
```
Comment (amrbashir) [inline plugins/window-state/src/lib.rs]: there is no need to use a static variable, you can just store it on the builder and clone/move it to the appropriate places, see https://github.com/tauri-apps/plugins-workspace/pull/546 for previous work.

**Review by amrbashir — CHANGES_REQUESTED** [review]
Alright, let's do this

File: plugins/window-state/src/lib.rs (hunk 1 above)
Comment (thewh1teagle) [reply plugins/window-state/src/lib.rs]: I restored the branch to latest v2 and [added these changes](https://github.com/thewh1teagle/plugins-workspace/commits/feat/window-state/custom-state-filename/), I think you need to open the PR, 
If it's not possible I'll create another PR

Comment (amrbashir) [reply plugins/window-state/src/lib.rs]: https://github.com/tauri-apps/plugins-workspace/pull/1138

**Pull request closed without merging**